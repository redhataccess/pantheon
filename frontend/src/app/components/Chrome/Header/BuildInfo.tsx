import React, { Component } from 'react';
import { Link } from "react-router-dom";

class BuildInfo extends Component {
    public state = {
        buildDate: '',
        commitHash: '',
        commitText: ''
    };

    public componentDidMount() {
        this.getBuildInfo()
    }

    public render() {
        return (
            <React.Fragment>
                <div className="column-view">
                    {!this.state.commitHash.includes("OPENSHIFT_BUILD_COMMIT") && <Link to=""
                        onClick={this.commitRedirect}>
                        {this.state.buildDate}
                    </Link>}
                    {this.state.commitHash.includes("OPENSHIFT_BUILD_COMMIT") && this.state.buildDate}
                </div>
            </React.Fragment>
        );
    }

    public getBuildInfo() {
        const backend = "/pantheon/builddate.json"
        if (this.state.buildDate === '') {
            fetch(backend)
                .then(response => response.json())
                .then(responseJSON => this.setState({ buildDate: "Build Date: " + responseJSON.buildDate, commitHash: responseJSON.commitHash }))
        }
    };

    private commitRedirect = () => {
        window.location.href = "https://github.com/redhataccess/pantheon/commit/" + this.state.commitHash
    }
}

export { BuildInfo }