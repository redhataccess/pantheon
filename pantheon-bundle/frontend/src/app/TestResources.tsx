import { IAppState } from './app';

const mockStateUser: IAppState = {
    isAdmin: false,
    isNavOpen: true,
    userAuthenticated: true,
    username: "demo"
}

const mockStateAdmin: IAppState = {
    isAdmin: true,
    isNavOpen: true,
    userAuthenticated: true,
    username: "admin"
}

const mockStateGuest: IAppState = {
    isAdmin: false,
    isNavOpen: true,
    userAuthenticated: false,
    username: "anonymous"
}

export { mockStateUser, mockStateAdmin, mockStateGuest }