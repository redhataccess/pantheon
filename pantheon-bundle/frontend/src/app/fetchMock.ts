if (!self.fetch) {
    self.fetch = (input: RequestInfo, init?: RequestInit | undefined) => {
        const p = new Promise<Response>((resolve, reject) => {
            // just don't do anything. this exists so that our tests don't try and actually resolve fetches,
            // (which is exactly what isomorphic-fetch tried to do while we were importing it...)
            // and calling nothing here allows us to skip all of our fetch.then() and fetch.catch() code.
        })
        return p
    }
}