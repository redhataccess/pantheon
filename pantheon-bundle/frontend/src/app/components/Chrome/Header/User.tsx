import React, { Component } from "react"
import { Link } from "react-router-dom"
import {
    Button, Dropdown,
    DropdownToggle,
    DropdownItem, Modal, DropdownPosition
} from "@patternfly/react-core"
import { HelpIcon } from "@patternfly/react-icons"
import "@app/app.css"
import { IAppState } from "@app/app"

interface IState {
    helpDropdownOpen: boolean
    loginUrl: string
    logoutUrl: string
}

class User extends Component<IAppState, IState> {
    constructor(props) {
        super(props)
        this.state = {
            helpDropdownOpen: false,
            loginUrl: "/auth/login",
            logoutUrl: "/system/sling/logout"
        }
    }

    public render() {
        const dropdownItems = [
            <DropdownItem key="userguide" href="/pantheon/docs/assemblies/assembly-pantheon-help.html" target="_blank">User Guide</DropdownItem>,
            <DropdownItem key="contribute" href="https://github.com/redhataccess/pantheon" target="_blank">Contribute to Pantheon</DropdownItem>
        ]
        return (
            <React.Fragment>
                <Dropdown onSelect={this.onHelpSelect}
                    toggle={
                        <DropdownToggle toggleIndicator={null} onToggle={this.onHelpToggle}>
                            <HelpIcon />
                        </DropdownToggle>
                    }
                    isPlain={true}
                    isOpen={this.state.helpDropdownOpen}
                    dropdownItems={dropdownItems}
                    position={DropdownPosition.right}
                />
                <a className="p2-header__login"
                    href={this.props.userAuthenticated ? this.state.logoutUrl : this.state.loginUrl}
                    onClick={this.conditionalRedirect}>
                    {this.props.userAuthenticated ? "[" + this.props.username + "]" : "Log In"}
                </a>
            </React.Fragment>
        )
    }

    private onHelpToggle = helpDropdownOpen => {
        this.setState({ helpDropdownOpen })
    }

    private onHelpSelect = () => {
        this.setState({
            helpDropdownOpen: !this.state.helpDropdownOpen
        })
    }

    private conditionalRedirect = () => {
        if (this.props.userAuthenticated) {
            fetch(this.state.logoutUrl)
                .then(response => window.location.href = "/pantheon")
        }
    }
}

export { User }
