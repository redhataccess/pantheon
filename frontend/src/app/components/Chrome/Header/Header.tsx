import React from 'react'
import { PageHeader } from '@patternfly/react-core'
import { Brand } from '../Header/Brand/Brand'
import { User }  from '../Header/User'
import { IAppState } from '@app/app'

export interface IHeaderProps {
    isNavOpen: boolean
    onNavToggle: any
    appState: React.PropsWithChildren<IAppState>
}

export const Header: React.FunctionComponent<IHeaderProps> = ({
    isNavOpen,
    onNavToggle,
    appState
}) => {
    return (
        <PageHeader
            logo={<Brand />}
            toolbar={<User {...appState} />}
            showNavToggle={true}
            isNavOpen={isNavOpen}
            onNavToggle={onNavToggle}
        />
    )
}