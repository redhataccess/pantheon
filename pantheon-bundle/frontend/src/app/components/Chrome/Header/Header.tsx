import React from 'react'
import { PageHeader, Brand } from '@patternfly/react-core'
// import { Brand } from '../Header/Brand/Brand'
import { User }  from '../Header/User'
import { IAppState } from '@app/app'
import logo from '../../../images/Pantheon2-logo-white.png';

export interface IHeaderProps {
    isNavOpen: boolean
    onNavToggle: any
    appState: React.PropsWithChildren<IAppState>
}

export const Header: React.FunctionComponent<IHeaderProps> = ({
    isNavOpen,
    onNavToggle,
    appState
}) => {
    return (
        <PageHeader
            logo={<Brand src={logo} alt="Pantheon" className="PageHeader__Brand" />}
            logoProps={
                {
                'href': '/',
                // @todo Replace in CSS file when a project stylesheet has been setup
                'style': {'maxWidth': '220px'}
                }
            }
            toolbar={<User {...appState} />}
            showNavToggle={true}
            isNavOpen={isNavOpen}
            onNavToggle={onNavToggle}
        />
    )
}