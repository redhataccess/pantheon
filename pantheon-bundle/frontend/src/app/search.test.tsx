import React from 'react'
import { Search } from '@app/search'
import { BuildInfo } from './components/Chrome/Header/BuildInfo'
import { HashRouter as Router, Link } from 'react-router-dom'
import { mount, shallow } from 'enzyme'
import { DataList, Button, TextInput, Level, LevelItem, Checkbox, Alert } from '@patternfly/react-core'
import renderer from 'react-test-renderer'
import sinon from "sinon"
import { mockStateUser } from '@app/TestResources'
import '@app/fetchMock'

describe('Search tests', () => {
  test('should render default Search component', () => {
    const view = shallow(<Search {...mockStateUser} />)
    expect(view).toMatchSnapshot()
  })

  it('should render a Button', () => {
    const wrapper = mount(<Router><Search {...mockStateUser} /></Router>)
    const button = wrapper.find(Button)
    expect(button.exists()).toBe(true)
  })

  it('should render a DataList', () => {
    const wrapper = mount(<Router><Search {...mockStateUser} /></Router>)
    const dataList = wrapper.find(DataList)
    expect(dataList.exists()).toBe(true)
  })

  it('should render a TextInput', () => {
    const wrapper = mount(<Router><Search {...mockStateUser} /></Router>)
    const textInput = wrapper.find(TextInput)
    expect(textInput.exists()).toBe(true)
  })

  it('should render a BuildInfo component', () => {
    const wrapper = mount(<Router><Search {...mockStateUser} /></Router>)
    const buildInfo = wrapper.find(BuildInfo)
    expect(buildInfo.exists()).toBe(true)
  })

  it('should render a level component', () => {
    const wrapper = mount(<Router><Search {...mockStateUser} /></Router>)
    const level = wrapper.find(Level)
    expect(level.exists()).toBe(true)
  })

  it('should render a Level component', () => {
    const wrapper = mount(<Router><Search {...mockStateUser} /></Router>)
    const level = wrapper.find(Level)
    expect(level.exists()).toBe(true)
  })

  it('should render a levelItem component', () => {
    const wrapper = mount(<Router><Search {...mockStateUser} /></Router>)
    const levelItem = wrapper.find(LevelItem)
    expect(levelItem.exists()).toBe(true)
  })

  it('should not render a dangerAlert for null or positive search results', () => {
    const wrapper = mount(<Router><Search {...mockStateUser} /></Router>)
    const dangerAlert = wrapper.find(Alert)
    expect(dangerAlert.exists()).toBe(false)
  })

  it('should handle state changes for displayLoadIcon', () => {
    const wrapper = shallow(<Router><Search {...mockStateUser} /></Router>)
    wrapper.setState({ 'displayLoadIcon': true })
    expect(wrapper.state('displayLoadIcon')).toBe(true)
    wrapper.setState({ 'displayLoadIcon': false })
    expect(wrapper.state('displayLoadIcon')).toBe(false)
  })

  it('should handle state changes for isSearchException', () => {
    const wrapper = shallow(<Router><Search {...mockStateUser} /></Router>)
    wrapper.setState({ 'isSearchException': false })
    expect(wrapper.state('isSearchException')).toBe(false)
    wrapper.setState({ 'isSearchException': true })
    expect(wrapper.state('isSearchException')).toBe(true)
  })

  it('should handle state changes for empty results for search', () => {
    const wrapper = shallow(<Router><Search {...mockStateUser} /></Router>)
    wrapper.setState({ 'isEmptyResults': false })
    expect(wrapper.state('isEmptyResults')).toBe(false)
    wrapper.setState({ 'isEmptyResults': true })
    expect(wrapper.state('isEmptyResults')).toBe(true)
  })

  it('should render a level component', () => {
    const wrapper = mount(<Router><Search {...mockStateUser} /></Router>)
    const level = wrapper.find(Level)
    expect(level.exists()).toBe(true)
  })


  it('test getRows function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.getRows).toMatchSnapshot()
  })

  it('test buildSearchUrl function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.buildSearchUrl).toMatchSnapshot()
  })

  it('test hideAlertOne function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.hideAlertOne).toMatchSnapshot()
  })

  it('test confirmDeleteOperation function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.confirmDeleteOperation).toMatchSnapshot()
  })

  it('test cancelDeleteOperation function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.cancelDeleteOperation).toMatchSnapshot()
  })

  it('test sortByUploadTime function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.sortByUploadTime).toMatchSnapshot()
  })

  it('test sortByDescription function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.sortByDescription).toMatchSnapshot()
  })

  it('test sortByName function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.sortByName).toMatchSnapshot()
  })

  it('test dismissNotification function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.dismissNotification).toMatchSnapshot()
  })

  it('test doSearch function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.doSearch).toMatchSnapshot()
  })

  it('test newSearch function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.newSearch).toMatchSnapshot()
  })

  it('test delete function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.delete).toMatchSnapshot()
  })

  it('test handleDeleteCheckboxChange function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.handleDeleteCheckboxChange).toMatchSnapshot()
  })

  it('test handleSelectAll function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.handleSelectAll).toMatchSnapshot()
  })

  it('test setInput function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.setInput).toMatchSnapshot()
  })

  it('test fetchTimeout function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.fetchTimeout).toMatchSnapshot()
  })

  it('test changePerPageLimit function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.changePerPageLimit).toMatchSnapshot()
  })

  it('test buildTransientPathArray function', () => {
    const wrapper = renderer.create(<Router><Search {...mockStateUser} /></Router>)
    const inst = wrapper.getInstance()
    expect(inst.buildTransientPathArray).toMatchSnapshot()
  })

})
