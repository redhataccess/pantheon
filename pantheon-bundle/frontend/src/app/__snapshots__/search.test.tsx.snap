// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests should render default Search component 1`] = `
<Fragment>
  <div>
    <div>
      <SearchFilter
        isSortedUp={true}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onSort={[Function]}
        value=""
      />
      <div
        className="notification-container"
      >
        <Pagination
          bottom={false}
          handleMoveLeft={[Function]}
          handleMoveRight={[Function]}
          handleMoveToFirst={[Function]}
          handlePerPageLimit={[Function]}
          nextPageRecordCount={1}
          pageNumber={1}
          perPageLimit={25}
          showDropdownOptions={true}
        />
      </div>
      <DataList
        aria-label="Simple data list"
      >
        <DataListItem
          aria-labelledby="simple-item1"
        >
          <DataListItemRow
            id="data-rows-header"
          >
            <Checkbox
              aria-label="controlled checkbox example"
              aria-labelledby="width-ex1-check1"
              checked={false}
              className="checkbox"
              id="check"
              isChecked={false}
              isDisabled={false}
              isValid={true}
              onChange={[Function]}
            />
            <DataListItemCells
              dataListCells={
                Array [
                  <DataListCell
                    width={2}
                  >
                    <button
                      aria-label="sort column by name"
                      className="sp-prop"
                      id="span-name"
                      onClick={[Function]}
                    >
                      Name
                    </button>
                  </DataListCell>,
                  <DataListCell
                    width={2}
                  >
                    <button
                      aria-label="sort column by description"
                      className="sp-prop"
                      id="span-name"
                      onClick={[Function]}
                    >
                      Description
                    </button>
                  </DataListCell>,
                  <DataListCell>
                    <span
                      className="sp-prop-nosort"
                      id="span-source-type"
                    >
                      Source Type
                    </span>
                  </DataListCell>,
                  <DataListCell>
                    <span
                      className="sp-prop-nosort"
                      id="span-source-name"
                    >
                      Source Name
                    </span>
                  </DataListCell>,
                  <DataListCell>
                    <button
                      aria-label="sort column by upload time"
                      className="sp-prop"
                      id="span-name"
                      onClick={[Function]}
                    >
                      Upload Time
                    </button>
                  </DataListCell>,
                ]
              }
            />
          </DataListItemRow>
          <DataListItemRow
            id="data-rows"
          />
          <Level
            gutter="md"
          >
            <LevelItem />
            <LevelItem>
              <div
                className="notification-container"
              >
                <br />
                <br />
                <img
                  alt="Spinlogo"
                  src="test-file-stub"
                />
                <br />
                <br />
              </div>
            </LevelItem>
            <LevelItem />
          </Level>
          <DataListItemRow
            id="data-rows"
          />
        </DataListItem>
      </DataList>
      <div
        className="notification-container"
      >
        <Pagination
          bottom={true}
          handleMoveLeft={[Function]}
          handleMoveRight={[Function]}
          handleMoveToFirst={[Function]}
          handlePerPageLimit={[Function]}
          nextPageRecordCount={1}
          pageNumber={1}
          perPageLimit={25}
          showDropdownOptions={false}
        />
        <BuildInfo />
      </div>
      <div
        className="alert"
      />
    </div>
  </div>
</Fragment>
`;

exports[`Search tests test buildSearchUrl function 1`] = `undefined`;

exports[`Search tests test buildTransientPathArray function 1`] = `undefined`;

exports[`Search tests test cancelDeleteOperation function 1`] = `undefined`;

exports[`Search tests test changePerPageLimit function 1`] = `undefined`;

exports[`Search tests test confirmDeleteOperation function 1`] = `undefined`;

exports[`Search tests test delete function 1`] = `undefined`;

exports[`Search tests test dismissNotification function 1`] = `undefined`;

exports[`Search tests test doSearch function 1`] = `undefined`;

exports[`Search tests test fetchTimeout function 1`] = `undefined`;

exports[`Search tests test getRows function 1`] = `undefined`;

exports[`Search tests test handleDeleteCheckboxChange function 1`] = `undefined`;

exports[`Search tests test handleSelectAll function 1`] = `undefined`;

exports[`Search tests test hideAlertOne function 1`] = `undefined`;

exports[`Search tests test newSearch function 1`] = `undefined`;

exports[`Search tests test setInput function 1`] = `undefined`;

exports[`Search tests test sortByDescription function 1`] = `undefined`;

exports[`Search tests test sortByName function 1`] = `undefined`;

exports[`Search tests test sortByUploadTime function 1`] = `undefined`;
